client/comms/wsconn.go

```diff
@@ -110,14 +111,15 @@ type responseHandler struct {
 
 // WsCfg is the configuration struct for initializing a WsConn.
 type WsCfg struct {
-	// URL is the websocket endpoint URL.
-	URL string
-
```

Comment:
Why does this need to be a separate argument to the constructor?

Reply:
I wanted to URL to only be stored in one place, be able to be updated, and I didn't want to add a mutex for the config or a field in the config. If we don't update the URL, when there is a read error, and it reconnects, it will only reconnect to the first market that was subscribed to.

Reply:
> I wanted to URL to only be stored in one place

I don't really see this as a compelling reason to change the function signature for every consumer. You can still have an unexported field that you update. When we have structs for config settings, we generally try to keep everything contained to the struct.